src/RpcNep11Tracker/RpcNep11Tracker.cs

```diff
@@ -0,0 +1,428 @@
+            {
+                // Executions that fault won't modify storage, so we can skip them.
+                if (appExecuted.VMState.HasFlag(VMState.FAULT)) continue;
+                foreach (var notifyEventArgs in appExecuted.Notifications)
+                {
+                    if (!(notifyEventArgs?.State is VmArray stateItems) || stateItems.Count == 0)
+                        continue;
+                    var contract = GetContract(snapshot, notifyEventArgs.ScriptHash);
+                    if (contract?.Manifest.SupportedStandards.Contains("NEP-11") == false) continue;
+                    HandleNotification(snapshot, notifyEventArgs.ScriptContainer, notifyEventArgs.ScriptHash, notifyEventArgs.EventName, stateItems, transfers, ref transferIndex);
+                }
+            }
+
+            var contracts = new Dictionary<UInt160, (bool isDivisible, ContractState state)>();
+
+            foreach (var transferRecord in transfers)
+            {
+                if (!contracts.ContainsKey(transferRecord.asset))
+                {
+                    var state = GetContract(snapshot, transferRecord.asset);
+                    var balanceMethod = state.Manifest.Abi.GetMethod("balanceOf", 1);
+                    var balanceMethod2 = state.Manifest.Abi.GetMethod("balanceOf", 2);
+                    if (balanceMethod == null && balanceMethod2 == null)
+                    {
+                        Log($"{state.Hash} is not nft!", LogLevel.Warning);
+                        continue;
+                    }
+                    var isDivisible = balanceMethod2 != null;
+                    contracts[transferRecord.asset] = (isDivisible, state);
+                }
+                var asset = contracts[transferRecord.asset];
+                if (asset.isDivisible)
+                {
+                    SaveDivisibleNFT(transferRecord, snapshot);
+                }
+                else
+                {
+                    SaveNFT(transferRecord);
+                }
+            }
+        }
+
+        private void SaveDivisibleNFT(TransferRecord record, DataCache snapshot)
+        {
+            using ScriptBuilder sb = new ScriptBuilder();
+            sb.EmitDynamicCall(record.asset, "balanceOf", record.from, record.tokenId);
+            sb.EmitDynamicCall(record.asset, "balanceOf", record.to, record.tokenId);
+            using ApplicationEngine engine = ApplicationEngine.Run(sb.ToArray(), snapshot, settings: neoSystem.Settings);
+            if (engine.State.HasFlag(VMState.FAULT) || engine.ResultStack.Count != 2)
+            {
+                Console.WriteLine($"Fault:{record.from},{record.to},{record.tokenId.GetSpan().ToHexString()}");
+                return;
+            }
+            Put(Nep11BalancePrefix, new Nep11BalanceKey(record.to, record.asset, record.tokenId), new Nep11Balance() { Balance = engine.ResultStack.Pop().GetInteger(), LastUpdatedBlock = _currentHeight });
```

Comment:
Is not easy to deny the service if the contract is prepared to return a wrong result? it will deny the service to any node that will have this plugin, I think that it should log the error but not deny the service, imagine if a consensus node has this plugin.

https://github.com/neo-project/neo-vm/blob/b48113f8dc787d059324c2d34c855f77b0e3645d/src/neo-vm/Types/StackItem.cs#L116

Reply:
Please check https://github.com/neo-project/neo-modules/pull/671